overlord/configstate/config/transaction.go

```diff
@@ -172,6 +172,14 @@ func (t *Transaction) Get(snapName, key string, result interface{}) error {
 	// commit changes onto a copy of pristine configuration, so that get has a complete view of the config.
 	config := t.copyPristine(snapName)
 	applyChanges(config, t.changes[snapName])
+
+	out := purgeNulls(config)
```

Comment:
Wouldn't it still work as before if you have this:
```
purgeNulls(config)
return getFromConfig(snapName, subkeys, 0, config, result)
```

Reply:
Yes, sure, note I'm not passing `out` to getFromConfig and the only reason for special casing this condition is to higlight the fact that that entire configuration was "reduced" to `nil` (by removing empty maps) while still in the transaction. We return `NoOptionError` in such case while maybe we should give 'snap has no configuration` error, so this special casing here would make handling it easy and obvious if we wanted to (but if we do that then we would need to change implementation of `GetFeatureFlag` which wants to report true or false if there is no config).

Nb, this special-case goes away with the change to not reduce empty maps per @pedronis comments.